### DISCUSSION TITLE
Timelapse with autoupload with counter

### DISCUSSION THREAD

**Discussion by MatthewPoland** [gopro/labs#308] [Q&A]
Hi, i'm trying to make a timelapse 15 min interval (48 photos a day) with autoupload, between 6AM - 6PM with GoPro Hero 9 or 8.
I want to capture the single photo and put camera into sleep to save power.
I have a hardware scheduler to turn USB power ON (06:00) and OFF (19:30) (with 4 additional power resets during the day).
As the GoPro is having problems when turning on and off frequently, I want it just to count time, to avoid clock problems.
I try to make a bunch of scripts and overwrite the boot script during the capture ... but with no luck :(
I tested command lines like so:

'1 Create special scripts:
```
!MWAKE=2
!MBOOT="!Lboot"
!SAVEboot="!Lstart"
!SAVEstart="mPN!S!SAVEboot="!Ltimelapse1"!300OR"
!SAVEupload="!U!SAVEboot="!Lstart"!OR"
```
'2 Create counter scripts
```
start >> mPN!S!SAVEboot="!Ltimelapse1"!300OR
timelapse1 >> mPN!S!SAVEboot="!Ltimelapse2"!300OR
timelapse2 >> mPN!S!SAVEboot="!Ltimelapse3"!300OR
...
timelapse49 >> mPN!S!SAVEboot="!Lupload"!OR
upload >> !U!SAVEboot="!Lstart"!OR
```
but unfortunately !SAVE option is saving everything even after ="" so with reboot command as well (!300OR)
so after the photo is taken nothing happen, and the boot script looks like for ex. `"!Ltimelapse1"!300OR`

My question is there any other way to separate SAVE command form saving everything outside "" ?

**Comment by dnewman-gpsw** [comment]
!SAVE should never been used within a script.  It is a just a QR Code way for adding scripts to the camera, which can also do directly by writing to the SD Card.  So redesign to not use !SAVE in this way.

**Reply by MatthewPoland** [reply]
I don't think is there any other way to store the lastest iteration that can survive power cycle. When power cycle GoPro it will boot with the lastest saved iteration saved into boot script. Otherwise it will starts from beginning. 

When using a !MBOOT the GoPro load script each time it wakes, doesn't matter it's from sleep :( or power cycle. It doesn't recognize rather it was sleep from repeat (above !9R) or power cycle !MWAKE=2 state.

Please correct me if I'm wrong.

**Reply by dnewman-gpsw** [reply]
Well those scripts won't work, but I'm having trouble understand how they solve the problem.

This is all you need:    
**dP>05:59<18:00!900SQ~!06:00S!1R**

hardware scheduler only needs to be turned off once a day, and only for a minute or so.  Having power more regular keeps the clock accurate.

But if you have power, why risk any of this complexity? Just use:
**!06:00NmTPp.10!S!43200E!U!1R**

The only downside is the stock Photo Timelapse doesn't have a 15s interval, so 10s is selected here.

The second script is far more reliable.

**Reply by MatthewPoland** [reply]
The problem is there is no grid power. 
We are using Voltaics power bank with solar panel.
Main reason why we don't want to use clock to schedule is that after power failure it's super unreliable.
Second, I don't think you ever used this code `dP>05:59<18:00!900SQ~!06:00S!1R` as a boot script.
If so, here is what will happen:

GoPro will boot-up and set into photo mode `dP`, next it will check the time condition `>05:59<18:00`, if true then go to sleep for 900sek. When it wakes with `alarm` or `power` it will boot and do same script from the beginning, looping over and over, never reach SQ to take even one photo.

So good luck with that.

If you do it as standard qr code command, it won't last power failure. 
What we usually do is:
dP>05:59<18:00!SQ+!900N~!06:00S!1R
It will work as boot script but GoPro like to wake up in the middle of the night and goes for sleep, or dont wake up at all randomly.
That's why we use hardware scheduler to power cycle.
But still when power dies, and it need day or more to recharge the clock condition is useless.
Our hardware scheduler has GSM time sync, and it can boot after long no power state.

Still don't have any solution for my problem.
I read that the new version H11 has the option of variables and storing them after restart, unfortunately I don't have GoPro 11 to check if you can program its behaviour in this way. Apparently there is an option to count the amount of !R and condition actions based on it.

**Reply by dnewman-gpsw** [reply]
Yes, only HERO11 at the moment. It is harder to make improvements on older cameras, I expect MAX and HERO10 will see some of these improvements, but older camera updates are uncertain.  Correct, none of these should ever be directly used in a boot script. This should work if power is somewhat reliable:

!MWAKE=2
!MBOOT="!Lboot"
!SAVEboot=dP<05:59!06:00R>18:00!06:00R!S!900R

Yes, if the power goes out for more than 12hours, the clock in camera will be wrong.  Except again on HERO11, you can use the GPS to sync the clock, and even have the script wait for GPS lock to make sure the time is correct.  If that power is that unreliable, why use a clock at all?

Just use:
!SAVEboot=mP!S!900R

Yes, you will have 2x the photos, but half will be in the hours you need.

Scheduling an upload without a clock will be harder. HERO11 supports a random variable, so you can get clockless upload, on ocassion.  On HERO8/9, use 

!SAVEboot=mP!S>18:00<18:15!U!900R

at it will be 6pm, at least once a day, even if the clock is wrong.
